src/sys/wait.rs

```diff
@@ -355,7 +355,10 @@ pub enum Id {
     PGid(Pid),
     /// Wait for the child referred to by the given PID file descriptor
     #[cfg(any(target_os = "android", target_os = "linux"))]
-    PIDFd(RawFd),
+    PIDFd(BorrowedFd<'fd>),
+    /// A helper variant to resolve the unused parameter (`'fd`) problem on platforms
+    /// other than Linux and Android.
+    _Unreachable(std::marker::PhantomData<&'fd std::convert::Infallible>),
```

Comment:
Put a [`std::convert::Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html) in it to make sure this variant could NEVER be constructed.

Reply:
Will mark this variant with a #[doc(hidden)].